.github/workflows/release-main-docker-image.yml

```diff
@@ -0,0 +1,40 @@
+name: Relase Main Docker image
+
+on:
+  push:
+    branches: [ main ]
+    
+jobs:
+  build_and_push_docker_image:
+    if: github.repository_owner == 'NethermindEth'
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        
+      # Log in to Docker Hub
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      # Build and push Docker images
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: |
+            nethermindeth/juno:${{github.run_number}}
+            nethermindeth/juno:latest
+
+      # Clean up environment
+      - name: Clean up environment
+        if: always()
+        run: |
+          rm -f ${HOME}/.docker/config.json
```

Comment:
When will this file run? Currently we don't want to publish any build from `main` branch

Reply:
when new commits are pushed to main